defs/build.bzl

```diff
@@ -20,10 +23,18 @@ def _gcs_upload_impl(ctx):
   return struct(
       runfiles = ctx.runfiles(
           files = ctx.files.data + ctx.files.uploader +
-            [ctx.version_file, ctx.outputs.targets]
+            [ctx.info_file, ctx.version_file, ctx.outputs.targets]
           )
       )
 
+# Adds an executable rule to upload the specified artifacts to GCS.
+#
+# The keys in upload_paths must match the elaborated targets exactly; i.e.,
+# one must specify "//foo/bar:bar" and not just "//foo/bar".
```

Comment:
Bazel seems to fully elaborate the targets before executing skylark rules, which makes sense.

It seems like buildifier (or something equivalent) prefers "//foo/bar" over "//foo/bar:bar", which means that we can't be consistent in BUILD files between the two lists.

I hope to get rid of `upload_paths` eventually, just using a dict in `data`.